src/Core/Echo.ControlFlow/ControlFlowNode.cs

```diff
@@ -281,6 +281,7 @@ public ControlFlowEdge<TInstruction> ConnectWith(ControlFlowNode<TInstruction> n
             var newBlock = new BasicBlock<TInstruction>(ParentGraph.Architecture.GetOffset(instructions[0]), instructions);
             var newNode = new ControlFlowNode<TInstruction>(newBlock.Offset, newBlock);
             ParentGraph.Nodes.Add(newNode);
+            newNode.ParentRegion = ParentRegion;
```

Comment:
Setting `ParentRegion` directly does not actually add it to the parent region (e.g. the number of nodes in `region.ProtectedRegion` in the test of this PR is still 1 after splitting). It is only used for bookkeeping and for an intuitive public API. 

For moving nodes into regions, directly add them to the region (e.g. using `ScopeRegion.Nodes.Add`) or use `ControlFlowNode.MoveToRegion` instead.